### PR TITLE
[Merged by Bors] - feat: port Data.Polynomial.AlgebraMap

### PR DESCRIPTION
**Pull request by j-loreaux**
- [x] depends on: #2817

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2817~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Polynomial/AlgebraMap.lean [hunk 1]
```diff
@@ -164,77 +167,87 @@ variable (x : A)
...
 
-@[simp]
+-- porting note: removed `@[simp]` because `simp` can prove this
 theorem aeval_zero : aeval x (0 : R[X]) = 0 :=
   AlgHom.map_zero (aeval x)
 #align polynomial.aeval_zero Polynomial.aeval_zero
 
-@[simp]
-theorem aeval_x : aeval x (X : R[X]) = x :=
+theorem aeval_X : aeval x (X : R[X]) = x :=
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/AlgebraMap.lean:201]: Where did `simp` go here?

File: Mathlib/Data/Polynomial/AlgebraMap.lean [hunk 2]
```diff
@@ -164,77 +167,87 @@ variable (x : A)
...
 theorem aeval_monomial {n : ℕ} {r : R} : aeval x (monomial n r) = algebraMap _ _ r * x ^ n :=
   eval₂_monomial _ _
 #align polynomial.aeval_monomial Polynomial.aeval_monomial
 
-@[simp]
-theorem aeval_x_pow {n : ℕ} : aeval x ((X : R[X]) ^ n) = x ^ n :=
-  eval₂_X_pow _ _
-#align polynomial.aeval_X_pow Polynomial.aeval_x_pow
+@[simp] -- porting note: simplifed the LHS because of `simpNF`
+theorem aeval_X_pow {n : ℕ} : aeval x (X : R[X]) ^ n = x ^ n := by
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/AlgebraMap.lean:218]: I think it would make morme sense to remove `simp` attribute and leave the original lemma as is.

Comment (j-loreaux) [reply Mathlib/Data/Polynomial/AlgebraMap.lean]: done

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**